server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -146,6 +149,10 @@ public static boolean isRemoteStoreAttributePresent(Settings settings) {
         return settings.getByPrefix(Node.NODE_ATTRIBUTES.getKey() + REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX).isEmpty() == false;
     }
 
+    public static boolean isRemoteStoreClusterStateEnabled(Settings settings) {
+        return RemoteClusterStateService.REMOTE_CLUSTER_STATE_ENABLED_SETTING.get(settings) && isRemoteStoreAttributePresent(settings);
```

Comment:
it is not checking remote state repo explicitly?

Reply:
this is not an issue since we are only allowing either all 3 (segment, translog, cluster state) to be set or none. Even for remote store checks we are just checking in the same way

Reply:
Remote cluster state is optional even for remote enabled nodes right?

Reply:
Please create an issue to decouple remote state repos and other repos based on remote state enabled flag.